### PR TITLE
TUP-609 Improve feedback when users reply to tickets

### PR DESCRIPTION
**Pull request by sophia-massie**
## Overview
The purpose of this task is to improve the UI of submitting a reply in the ticket modal. Upon submitting a reply, there should be a InlineMessage type="success" at the bottom-left of the modal that says
Your reply has been sent.

[This message should appear in a "modal footer"](https://xd.adobe.com/view/db2660cc-1011-4f26-5d31-019ce87c1fe8-ad17/screen/82d25240-6e20-491e-9f90-f7c9d534965a/specs/) (perhaps `<ModalFooter>`) so message visibility is not dependent on scroll.

Ideally, the tickets scroll in their own panel, like on CEP's ticket modal.

## Related

- [TUP-609](https://jira.tacc.utexas.edu/browse/TUP-609)

## Changes

- Added modal footer with inline messages on success of reply 
- Added success state to ticket reply form element
- Disable reply button when the modal has a successful reply state

## Testing

1. Select a ticket on either the dashboard or the ticket page
2. Submit a reply
3. Check that the success message appears on successful submit and the Reply button is disabled.

## UI
<h3>On success </h3>

Screenshot 2023-11-21 at 5 07 57 PM

<h3>On Error</h3>

Screenshot 2023-11-21 at 5 07 30 PM

### CODE REVIEW

**Review by wesleyboar — CHANGES_REQUESTED** [review]
Thank you. This is acceptable. I have requests.

<details><summary>1. Remove <code>&lt;ModalFooter&gt;</code> or attempt to place <a href="https://react-bootstrap.netlify.app/docs/components/modal/#static-markup" target="_blank">as Reactstrap / Bootstrap intends</a>.</summary>

The `<ModalFooter>` is intended to be at the bottom of the modal (beneath [`<ModalBody>`](https://github.com/TACC/tup-ui/blob/v1.1.2/libs/tup-components/src/tickets/TicketCreateModal/TicketCreateModal.tsx#L50-L52)), instead of within the ticket reply panel (thus within `<ModalBody>`).

But given the architecture, that seams cumbersome, so I don't mind continuing to not use `<ModalFooter>`. I do not want a future dev to assume `<ModalFooter>` is allowable in `<ModalBody>`.

</details>

<details><summary>2. Watch for a response to <a href="https://tacc-team.slack.com/archives/CQUA4D5KJ/p1700682176767029?thread_ts=1700682029.516199&cid=CQUA4D5KJ" target="_blank">my request to V.T. to let button remain enabled</a>.</summary>

Just because a client or stakeholder asks for something, does not mean we do it. Let's evaluate the solution in the context of architecture, UX, design, business, et cetera. I let her see Design's idea that success message may be enough to resolve her reason for suggesting disabling button.

_P.S. Design had considered sorting messages too, so user sees new message appear, but that order deviates from RT UI, so needs more consideration._

</details>

**Comment by sophia-massie** [comment]
> Thank you. This is acceptable. I have requests.
> 
> 1. Remove `<ModalFooter>` or attempt to place [as Reactstrap / Bootstrap intends](https://react-bootstrap.netlify.app/docs/components/modal/#static-markup).
> The `<ModalFooter>` is intended to be at the bottom of the modal (beneath [`<ModalBody>`](https://github.com/TACC/tup-ui/blob/v1.1.2/libs/tup-components/src/tickets/TicketCreateModal/TicketCreateModal.tsx#L50-L52)), instead of within the ticket reply panel (thus within `<ModalBody>`).
> 
> But given the architecture, that seams cumbersome, so I don't mind continuing to not use `<ModalFooter>`. I do not want a future dev to assume `<ModalFooter>` is allowable in `<ModalBody>`.
> 
> 2. Watch for a response to [my request to V.T. to let button remain enabled](https://tacc-team.slack.com/archives/CQUA4D5KJ/p1700682176767029?thread_ts=1700682029.516199&cid=CQUA4D5KJ).
Ah sorry I thought I moved the footer out of the body thanks for catching that. 

I will keep an eye on VTs response and will defer to design. Thanks for the detailed notes in the subtasks on Jira!

File: libs/tup-components/src/tickets/TicketDetailModal/TicketReplyForm.tsx [hunk 1]
```diff
@@ -44,6 +44,11 @@ export const TicketReplyForm: React.FC<{ ticketId: string }> = ({
     if (values.status) formData.append('status', values.status);
     mutate(formData, {
       onSuccess: () => resetForm(),
+      onSettled: () => {
+        setTimeout(() => {
+          mutation.reset();
```
Comment (wesleyboar) [inline libs/tup-components/src/tickets/TicketDetailModal/TicketReplyForm.tsx:49]: Slick. [Resetting the mutation](https://tanstack.com/query/v3/docs/react/guides/mutations#resetting-mutation-state) seems like "the right way" to me.

**Review by wesleyboar — APPROVED** [review]
Works. Code looks good.

**Pull request merged**